Framework/Beamline/src/DetectorInfo.cpp

```diff
@@ -360,15 +368,26 @@ DetectorInfo::buildMergeIndices(const DetectorInfo &other) const {
   // TODO If we make masking time-independent we need to check masking here.
 
   if (m_isSyncScan) {
-    for (const auto &interval1 : *other.m_scanIntervals) {
-      for (const auto &interval2 : *m_scanIntervals) {
-        if (!((interval1.second <= interval2.first) ||
-              (interval1.first >= interval2.second))) {
-          failMerge("scan intervals overlap in sync scan");
+    std::vector<bool> merge(other.m_scanIntervals->size(), true);
+
+    for (size_t t1 = 0; t1 < other.m_scanIntervals->size(); ++t1) {
+      for (size_t t2 = 0; t2 < m_scanIntervals->size(); ++t2) {
+        if ((*other.m_scanIntervals)[t1] == (*m_scanIntervals)[t2]) {
+          for (size_t detIndex = 0; detIndex < size(); ++detIndex) {
+            const size_t linearIndex1 = linearIndex({detIndex, t1});
+            const size_t linearIndex2 = linearIndex({detIndex, t2});
+            checkIdentitcalIntervals(other, linearIndex1, linearIndex2);
+          }
+          merge[t1] = false;
+        } else if (!(((*other.m_scanIntervals)[t1].second <=
+                      (*m_scanIntervals)[t2].first) ||
+                     ((*other.m_scanIntervals)[t1].first >=
+                      (*m_scanIntervals)[t2].second))) {
```

Comment:
This comparison looks slightly different than in the non-sync case, should it be identical?

Reply:
Yes, it should, previously this was to check identical or overlap, now it should look the same.